state/secrets.go

```diff
@@ -1027,12 +1059,15 @@ func (st *State) removeSecretConsumer(uri *secrets.URI) error {
 	refCountsCollection, closer := st.db().GetCollection(refcountsC)
 	defer closer()
 	for _, doc := range docs {
-		key := secretConsumerLabelKey(doc.ConsumerTag, doc.Label)
-		_, err = refCountsCollection.Writeable().RemoveAll(bson.D{{
-			"_id", key,
-		}})
+		keys := []string{
+			secretConsumerLabelKey(doc.ConsumerTag, doc.Label),
+			secretOwnerLabelKey(doc.ConsumerTag, doc.Label),
```

Comment:
I don't quite understand why we need to consider owner labels when removing a secret consumer?

Reply:
Because now we create a refCount for both the consumer and owner whenever the unit sets an owner label or consumer label. So we have to cleanup both of them.